Fix failure to download gnu units on mike

```bash
bash ./cloud/general/init-gnu-units.sh /work/jgflemin/mike3/production/asgs/opt gfortran 4
--2024-05-26 07:45:16--  https://ftp.gnu.org/gnu/units/units-2.21.tar.gz
Resolving ftp.gnu.org (ftp.gnu.org)... 209.51.188.20, 2001:470:142:3::b
Connecting to ftp.gnu.org (ftp.gnu.org)|209.51.188.20|:443... connected.
GnuTLS: Error in the pull function.
Unable to establish SSL connection.
tar (child): ./units-2.21.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
./cloud/general/init-gnu-units.sh: line 35: cd: units-2.21: No such file or directory
./cloud/general/init-gnu-units.sh: line 36: ./configure: No such file or directory
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.
post condition for "Step for installing units" FAILED, stopping. Please fix and rerun.
```

--- a/cloud/general/init-gnu-units.sh
+++ b/cloud/general/init-gnu-units.sh
@@ -19,13 +19,13 @@ if [ "$COMPILER" == "clean" ]; then
   exit
 fi
 
-UNITS_VERSION=2.21
+UNITS_VERSION=2.23
 UNITS_DIR=units-${UNITS_VERSION}
 UNITS_TGZ=${UNITS_DIR}.tar.gz
 cd $_ASGS_TMP
 
 if [ ! -e ${UNITS_TGZ} ]; then
-  wget --no-check-certificate https://ftp.gnu.org/gnu/units/${UNITS_TGZ}
+  wget --no-check-certificate https://ftpmirror.gnu.org/units/${UNITS_TGZ}
 fi
 
 rm -rf ./$UNITS_DIR 2> /dev/null

